### PR TITLE
fix(card): Remove overflow:hidden

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #4980

## Summary

fix(card): Remove overflow:hidden. (#4980)

### CODE REVIEW

File: src/components/card/card.scss [hunk 1]
```diff
@@ -110,6 +109,8 @@
 
 .thumbnail-wrapper {
   @apply flex;
+  border-start-start-radius: var(--calcite-border-radius-base);
+  border-start-end-radius: var(--calcite-border-radius-base);
```
Comment (mjuniper) [inline src/components/card/card.scss:113]: This is the obvious solution but I think it breaks down in a case like this: https://codepen.io/mjuniper/pen/qBoNJpv?editors=1100.

Comment (driskull) [reply src/components/card/card.scss:113]: yeah. It does :\

Comment (driskull) [reply src/components/card/card.scss:113]: We might just have to leave it up to the user to apply a radius on their slotted image. Thoughts @jcfranco ?

Comment (mjuniper) [reply src/components/card/card.scss:113]: This case makes it kinda gnarly and I have not been able to find a solution yet.

Comment (mjuniper) [reply src/components/card/card.scss:113]: I thought maybe clip-path might help us here but I have not been able to make that work.

Comment (driskull) [reply src/components/card/card.scss]: We might just have to round all corners

Comment (jcfranco) [reply src/components/card/card.scss]: I'll defer to @macandcheese on this one.

**Pull request closed without merging**